src/OpenTracingActiveRecordBehavior.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+namespace Websupport\OpenTracing;
+
+use CEvent;
+use CModelEvent;
+use OpenTracing\Scope;
+
+/**
+ * Class OpenTracingActiveRecordBehavior
+ * @package Websupport\OpenTracing
+ *
+ * @method \CActiveRecord getOwner()
+ */
+class OpenTracingActiveRecordBehavior extends \CActiveRecordBehavior
+{
+    /** @var string */
+    public $opentracingId = 'opentracing';
+
+    /**
+     * Whether we should hook to before and after find methods.
+     * Be careful with this option. There is option, where you can disable calling afterFind method
+     * directly in the code.
+     * @var bool
+     */
+    public $traceFind = false;
+
+    /**
+     * Whether we should hook to before and after save methods.
+     * @var bool
+     */
+    public $traceSave = true;
+
+    /**
+     * Whether we should hook to before and after delete methods.
+     * @var bool
+     */
+    public $traceDelete = true;
+
+    /** @var ?Scope */
+    private $activeScope;
+
+    public function attach($owner)
+    {
+        parent::attach($owner);
+
+        if (!\Yii::app()->hasComponent($this->opentracingId)) {
+            $this->setEnabled(false);
+        }
+    }
+
+    /**
+     * Responds to {@link CActiveRecord::onBeforeSave} event.
+     * Override this method and make it public if you want to handle the corresponding
+     * event of the {@link CBehavior::owner owner}.
+     * You may set {@link CModelEvent::isValid} to be false to quit the saving process.
+     * @param CModelEvent $event event parameter
+     * @throws
+     */
+    public function beforeSave($event)
+    {
+        if (!$this->traceSave) {
+            return;
+        }
+
+        $this->activeScope = \Yii::app()->getComponent($this->opentracingId)->startActiveSpan(
+            $this->spanName($this->getOwner()->getIsNewRecord() ? 'INSERT' : 'UPDATE'),
+            $this->spanOptionsFromActiveRecordClass()
+        );
+        $this->activeScope->getSpan()->log([
+            'attributes' => $this->getOwner()->getAttributes()
+        ]);
+    }
+
+    /**
+     * Responds to {@link CActiveRecord::onAfterSave} event.
+     * Override this method and make it public if you want to handle the corresponding event
+     * of the {@link CBehavior::owner owner}.
+     * @param CEvent $event event parameter
+     */
+    public function afterSave($event)
+    {
+        if (!$this->traceSave && $this->activeScope) {
+            return;
+        }
+        $this->activeScope->getSpan()
+            ->setTag('db.active_record.primary_key', json_encode($this->getOwner()->getPrimaryKey()));
```

Comment:
this can be done without json with just simple `implode(',', (array) $this->getOwner()->getPrimaryKey())`. this way it also match notation of composite primary keys

Reply:
also it should be moved to own method and reused in `beforeDelete`

Reply:
changed